lib/vector-core/src/event/discriminant.rs

```diff
@@ -158,6 +160,18 @@ fn hash_null<H: Hasher>(hasher: &mut H) {
     hasher.write_u8(0);
 }
 
+impl Display for Discriminant {
+    fn fmt(&self, fmt: &mut Formatter<'_>) -> Result<(), fmt::Error> {
+        for (i, value) in self.values.iter().enumerate() {
+            if i != 0 {
+                write!(fmt, "-")?;
+            }
+            write!(fmt, "{}", value.as_ref().map_or("none".to_string(), ToString::to_string))?
```

Comment:
This introduces a needless allocation, particularly since. I don't know how performance sensitive this actually is.
```suggestion
            if let Some(value) = value {
                value.fmt(fmt)?; // Basically just `write!(fmt, "{value}")`
            } else {
                fmt.write_str("none")?;
            }
```